pandas/io/parsers.py

```diff
@@ -2336,6 +2335,9 @@ def __init__(self, f: Union[FilePathOrBuffer, List], **kwds):
             if self.index_names is None:
                 self.index_names = index_names
 
+        if not hasattr(self, "_col_indices"):
```

Comment:
Why can't we *always* define this way on L2310 or L2297?

Reply:
We do not have acces to self.columns in L2297. Sometimes we set this in ``_infer_columns`` in L2302, so we have to check if it was already set to not override. Could move it up a bit but would have to check if it exists nevertheless

Reply:
can you instead define _col_list = None then as the default

Reply:
This would work of course, but we would get the mypy problems in again. Could do that nevertheless if this is preferable

Reply:
yes i think its important that this is *always* defined. you can use `Optional[List[int]]`, then you assert its not None

Reply:
Optional[List[int]] unfortunately raises the mypy error too. assert is not None does not help, so added the ignores back in